cg/cli/demultiplex/copy_novaseqx_data.py

```diff
@@ -0,0 +1,76 @@
+from pathlib import Path
+import os
+import shutil
+from typing import List, Optional
+
+from cg.constants.demultiplexing import DemultiplexingDirsAndFiles
+
+
+def is_copied(analysis_directory: Path) -> bool:
+    return Path(analysis_directory, DemultiplexingDirsAndFiles.COPY_COMPLETE).exists()
+
+
+def is_analyzed(analysis_directory: Path) -> bool:
+    return Path(
+        analysis_directory,
+        DemultiplexingDirsAndFiles.DATA,
+        DemultiplexingDirsAndFiles.ANALYSIS_COMPLETED,
+    ).exists()
+
+
+def is_in_demultiplexed_runs(flow_cell_name: str, demultiplexed_runs: Path) -> bool:
+    return Path(demultiplexed_runs, flow_cell_name).exists()
+
+
+def get_latest_analysis_directory(flow_cell: Path) -> Optional[Path]:
+    analysis_path = Path(flow_cell, DemultiplexingDirsAndFiles.ANALYSIS)
+
+    if not analysis_path.exists():
+        return None
+    analysis_versions = get_sorted_analysis_versions(analysis_path)
+    return analysis_versions[0] if analysis_versions else None
+
+
+def get_sorted_analysis_versions(analysis_path: Path) -> List[Path]:
```

Comment:
nice🔥

Reply:
a dcostring explaining the lambda function is beneficial here

Reply:
I think that is an implementation detail the caller does not need to know, but maybe a comment would be beneficial or refactoring it to be more readable

Reply:
Refactored it as to not use lambda functions